src/sentry/api/endpoints/project_details.py

```diff
@@ -132,6 +140,31 @@ def validate_uniform_sampling_rule(self, rules):
                 "Last rule is reserved for uniform rule which must have no conditions"
             )
 
+    def _contains_condition_rules(self, rules) -> bool:
+        for rule in rules:
+            if len(rule["condition"]["inner"]) > 0:
+                return True
+
+        return False
+
+    def authorize_dynamic_sampling(
+        self, organization, user, rules
+    ) -> DynamicSamplingAuthorizationResult:
+        is_basic = features.has("organizations:dynamic-sampling-basic", organization, actor=user)
+        is_advanced = features.has(
+            "organizations:dynamic-sampling-advanced", organization, actor=user
+        )
+        contains_condition_rules = self._contains_condition_rules(rules)
```

Comment:
What does this check do?

Reply:
I check whether the set of rules contain a condition rule, that is there exist at least a rule which has come conditions.

Reply:
Isn't this https://github.com/getsentry/sentry/pull/39756/files#diff-5999c0ed2577dc8b01a8dc3f0e7bd8749f46d34e06562b9d3582319e59cf85eeR125-R128 doing that prior to your function call?

Reply:
I am checking for the existence of a condition rule, the other functions are checking for the validity of the schema of rules.